system_metrics_collector/src/system_metrics_collector/proc_cpu_data.hpp

```diff
@@ -89,6 +90,46 @@ class ProcCpuData
   std::array<uint64_t, static_cast<int>(ProcCpuStates::kNumProcCpuStates)> times{};
 };
 
+class ProcPidCpuData
```

Comment:
This file is called `proc_cpu_data.hpp` because it contains `ProcCpuData`.
This should be in `proc_pid_cpu_data.hpp`.

Generally we have one class per header.